geomstats/geometry/lie_group.py

```diff
@@ -255,22 +253,14 @@ def exp(self, tangent_vec, base_point=None, point_type=None):
         result : array-like, shape=[n_samples, {dimension,[n,n]}]
             The exponentiated tangent vector
         """
-        if point_type is None:
-            point_type = self.default_point_type
-
         identity = self.get_identity(point_type=point_type)
-        identity = self.regularize(identity, point_type=point_type)
+        identity = gs.to_ndarray(
+            identity,
+            to_ndim=geomstats.vectorization.POINT_TYPES_TO_NDIMS[point_type])
         if base_point is None:
             base_point = identity
         base_point = self.regularize(base_point, point_type=point_type)
 
-        if point_type == 'vector':
-            tangent_vec = gs.to_ndarray(tangent_vec, to_ndim=2)
-            base_point = gs.to_ndarray(base_point, to_ndim=2)
-        if point_type == 'matrix':
-            tangent_vec = gs.to_ndarray(tangent_vec, to_ndim=3)
```

Comment:
What about the case of one matrix as tangent vector? You would have n_tangent_vec = n > 1 here?

Reply:
See above comment: the next PR will be on lie_groups and invariant_metric